### DISCUSSION TITLE
Setting the wrong font family silently fails and the default font is used instead

### DISCUSSION THREAD

**Discussion by feketegy** [ghostty-org/ghostty#4066] [Q&A]
When I list out the available fonts with `ghostty +list-fonts` I get a listing something like this:

![image](https://github.com/user-attachments/assets/cadaaf93-a3cc-45a4-b076-ea199b416b9c)

If I set the following config `font-family = "SF Mono Regular"` for the preferred font, Ghostty can't find it and silently reverts back to the default font which is _JetBrains Mono_.

I have to set `font-family = "SF Mono"` in order for Ghostty to pick it up and use it. 

I think this is a little bit confusing and not consistent because for setting bold, italic and bold italic then I have to set it like: 

```
font-family-bold = "SF Mono Bold"
font-family-italic = "SF Mono Regular Italic"
font-family-bold-italic = "SF Mono Bold Italic"
```

Also, if I would like to use, for example, _Semibold_ as the default weight, instead of _Regular_, I can't set it like `font-family = "SF Mono Semibold"` because this will revert back to the default font.

I guess I would have to set it like...

```
font-family = "SF Mono"
font-style = "Semibold"
```

...which is, again, a little bit confusing, I believe.

**Comment by rhodes-b** [comment]
This is the currently expected behavior due to font-fallback (on why it silently fails). Fonts may not include many symbols (especially unicode ones) so ghostty will attempt to fallback to either the default font or a font it thinks has good features.

You can see which font a symbol comes from with
`ghostty +show-face --string="X"` where x is whatever symbol you want to use to verify font (if you use some symbol your font does not support it will try to fallback to another font)
